when running event.send (or any fire_master), dont destroy io_loop

mattp- opened the pull request:
### What does this PR do?
without this param, the SaltEvent object closes the shared io_loop running the
minion itself, causing a hang.

### Previous Behavior
in multimaster, the minion itself will hang if this occurred, causing minion unresponsiveness.


### New Behavior
minion handles connection loss / event.send behavior properly.

### Tests written?
no

### Commits signed with GPG?
no

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

mattp- committed:
when running event.send (or any fire_master), dont destroy io_loop

without this param, the SaltEvent object closes the shared io_loop running the
minion itself, causing a hang.

mattp- commented:
this might be worthy of a backport to each release branch given the inertness vs the bug fix.

mattp- committed:
add a pylint disable for this warning

I don't know why, but when I add the passthrough kwarg to MinionEvent to Minion
when the destructor fires the bool is set to false incorrectly. py bug? my
ignorance? not sure.

mattp- committed:
modify event.send failure semantics slightly

it seems the current way to set failure was set ret to false on exception, but
continue to the next master. This potentially clobbers a failure. This change
ensures True ret if at least one master received the event, else false.

cachedout commented:
Nice fix here, @mattp-

mattp- commented:
please don't merge yet - dealing with some funky behavior with MinionEvent I'm trying to sort out. thanks

gtmanfred commented:
@mattp- i added [WIP] to the title, so it will mark the WIP job as in progress, once you are done if you remove that, we will check again.

Thanks!
Daniel

mattp- committed:
improve multimaster zeromq disconnection handling

we hook into SaltReqTimeoutError exceptions to trigger a status.master check;
from there some changes were made to the disconnected/connected event handling
so that the polling for restoration still occurs. It seems
master_alive_interval can be used to _always_ poll, so some changes around that
optional behavior were needed.

mattp- committed:
fix bug in zeromq transport with multimaster

when instantiating the singleton for a reqchannel a kwarg override can be
provided but it was not being used in the constructor.

mattp- committed:
event.send: don't disconnect io_loop when using MinionEvent

When using event.send on a running minion it tries to invoke the existing
io_loop, but by default was closing it on completion. this stops that from
happening.

mattp- committed:
Merge branch 'develop' into fix-event-send

cachedout commented:
@DmitryKuzmenko Can you dive into this as well, please?

mattp- committed:
some lint tidying

DmitryKuzmenko commented on salt/modules/event.py:
```diff
@@ -90,14 +90,16 @@ def fire_master(data, tag, preload=None):
-            return salt.utils.event.MinionEvent(__opts__, listen=False).fire_event(
-                {'data': data, 'tag': tag, 'events': None, 'pretag': None}, 'fire_master')
+            me = salt.utils.event.MinionEvent(__opts__, listen=False, keep_loop=True)
+            me.fire_event({'data': data, 'tag': tag, 'events': None, 'pretag': None}, 'fire_master')
```
@mattp- you've removed the `return` statement here. Just missed it? If not could you please comment this?

DmitryKuzmenko reviewed (approved):
I'm OK here with one minor question.

mattp- committed:
add back ret val accidentally removed
